config/routes.rb

```diff
@@ -1,506 +1,515 @@
+    get "/shell_bottom" => "shell#bottom"
+
+    get "/new" => "articles#new"
+    get "/new/:template" => "articles#new"
+
+    get "/pod", to: "podcast_episodes#index"
+    get "/podcasts", to: redirect("pod")
+    get "/readinglist" => "reading_list_items#index"
+    get "/readinglist/:view" => "reading_list_items#index", :constraints => { view: /archive/ }
+
+    get "/feed" => "articles#feed", :as => "feed", :defaults => { format: "rss" }
+    get "/feed/tag/:tag" => "articles#feed", :as => "tag_feed", :defaults => { format: "rss" }
+    get "/feed/:username" => "articles#feed", :as => "user_feed", :defaults => { format: "rss" }
+    get "/rss" => "articles#feed", :defaults => { format: "rss" }
+
+    get "/tag/:tag" => "stories#index"
+    get "/t/:tag", to: "stories#index", as: :tag
+    get "/t/:tag/edit", to: "tags#edit"
+    get "/t/:tag/admin", to: "tags#admin"
+    patch "/tag/:id", to: "tags#update"
+    get "/t/:tag/top/:timeframe" => "stories#index"
+    get "/t/:tag/page/:page" => "stories#index"
+    get "/t/:tag/:timeframe" => "stories#index",
+        :constraints => { timeframe: /latest/ }
+
+    get "/badge/:slug" => "badges#show"
+
+    get "/top/:timeframe" => "stories#index"
+
+    get "/:timeframe" => "stories#index", :constraints => { timeframe: /latest/ }
+
+    get "/:username/series" => "collections#index", :as => "user_series"
+    get "/:username/series/:id" => "collections#show"
+
+    # Legacy comment format (might still be floating around app, and external links)
+    get "/:username/:slug/comments" => "comments#index"
+    get "/:username/:slug/comments/:id_code" => "comments#index"
+    get "/:username/:slug/comments/:id_code/edit" => "comments#edit"
+    get "/:username/:slug/comments/:id_code/delete_confirm" => "comments#delete_confirm"
+
+    # Proper link format
+    get "/:username/comment/:id_code" => "comments#index"
+    get "/:username/comment/:id_code/edit" => "comments#edit"
+    get "/:username/comment/:id_code/delete_confirm" => "comments#delete_confirm"
+    get "/:username/comment/:id_code/mod" => "moderations#comment"
+    get "/:username/comment/:id_code/settings", to: "comments#settings"
+
+    get "/:username/:slug/:view" => "stories#show",
+        :constraints => { view: /moderate/ }
+    get "/:username/:slug/mod" => "moderations#article"
+    get "/:username/:slug/actions_panel" => "moderations#actions_panel"
+    get "/:username/:slug/manage" => "articles#manage"
+    get "/:username/:slug/edit" => "articles#edit"
+    get "/:username/:slug/delete_confirm" => "articles#delete_confirm"
+    get "/:username/:slug/stats" => "articles#stats"
+    get "/:username/:view" => "stories#index",
+        :constraints => { view: /comments|moderate|admin/ }
+    get "/:username/:slug" => "stories#show"
+    get "/:sitemap" => "sitemaps#show",
+        :constraints => { format: /xml/, sitemap: /sitemap-.+/ }
+    get "/:locale" => "stories#index", :constraints => { locale: LOCALE_LANG_REGEXP }
```

Comment:
Added this line to catch future routes such as `https://forem.dev/lang-fr-ca`.

Reply:
We actually _do_ allow `-` for orgs and pages which share the top level domain... We... So maybe we should add a validation so they can't start with `lang-`?

Reply:
Ohhh barnacles. We'll have to add validations from here forward to anything else that does is in the future.  I wonder if there's a good way to remind us this validation is needed in those cases 🤔

Reply:
Best is probably a page in documentation and/or comments explaining the general concerns and issues with the top level domain in general.

I think the best way to not mess this up is to have an acute awareness of the problems presented by using the same namespace for so many things.